device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsSessionDeviceOperation.java

```diff
@@ -337,6 +337,11 @@ boolean onLinkRemoteOpen(String linkName)
             {
                 if (entry.getValue().onLinkRemoteOpen(linkName))
                 {
+                    if (entry.getKey() == MessageType.DEVICE_TWIN)
+                    {
+                        sendMessage(entry.getValue().buildSubscribeToDesiredPropertiesProtonMessage(), entry.getKey(), deviceClientConfig.getDeviceId());
```

Comment:
If you make the '''buildSubscribeToDesiredPropertiesProtonMessage''' method concrete only on the twin class, you can safely cast entry.getValue() as a AmqpsDeviceTwin class to access it here